### PR TITLE
Replace closures in test packages

### PR DESCRIPTION
**Pull request by yuanchen8911**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR replaces all anonymous functions that convert `int32` to pointer with`Int32` and deprecated `Int32Ptr` with `Int32` in test packages. 

The use of the anonymous functions in these cases are unnecessary and hard to read. The change will improve code readability. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by pohly — APPROVED** [review]
This is a fairly small PR. Testing and reviewing a PR has a fixed cost, so the cost/benefit ratio is poor for small PRs.

Can you try to combine several cleanup commits in one PR? Each commit should be the same change, potentially in different files.

For example, this code pattern also occurs here:

```
test/e2e/framework/rc/rc_utils.go:                      Replicas: func(i int32) *int32 { return &i }(replicas),
```

**Comment by yuanchen8911** [comment]
> This is a fairly small PR. Testing and reviewing a PR has a fixed cost, so the cost/benefit ratio is poor for small PRs.
> 
> Can you try to combine several cleanup commits in one PR? Each commit should be the same change, potentially in different files.
> 
> For example, this code pattern also occurs here:
> 
> ```
> test/e2e/framework/rc/rc_utils.go:                      Replicas: func(i int32) *int32 { return &i }(replicas),
> ```

Replace all closures `func (int32 i) *int32` with `pointer.Int32Ptr` in test package.

File: test/e2e/apimachinery/resource_quota.go [hunk 1]
```diff
@@ -1992,7 +1993,7 @@ func newTestReplicationControllerForQuota(name, image string, replicas int32) *v
 			Name: name,
 		},
 		Spec: v1.ReplicationControllerSpec{
-			Replicas: func(i int32) *int32 { return &i }(replicas),
+			Replicas: pointer.Int32Ptr(replicas),
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1996]: I just remembered that `Int32Ptr` [is deprecated](https://github.com/kubernetes/utils/blob/a5ecb0141aa5b1b224e745aca203afdfd22d8b3a/pointer/pointer.go#L79-L82). Please use `Int32` instead.

**Review by pohly — CHANGES_REQUESTED** [review]

File: test/e2e/apimachinery/resource_quota.go (hunk 1 above)
Comment (yuanchen8911) [reply test/e2e/apimachinery/resource_quota.go]: Replaced `Int32Ptr` with `Int32`.

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Pull request merged**